Maple2.Server.DebugGame/Graphics/Scene/FollowCameraController.cs

```diff
@@ -0,0 +1,174 @@
+
+    public Vector3 CameraTarget { get; private set; } = Vector3.Zero;
+
+    // Camera properties that delegate to Camera.Transform
+    public Vector3 CameraPosition => Camera.Transform.Position;
+    public Quaternion CameraRotation => Camera.Transform.Quaternion;
+
+    public FollowCameraController(Camera camera) {
+        Camera = camera;
+    }
+
+    /// <summary>
+    /// Starts following a specific player
+    /// </summary>
+    public void StartFollowingPlayer(long playerId) {
+        IsFollowingPlayer = true;
+        FollowedPlayerId = playerId;
+        HasManuallyStopped = false; // Reset manual stop flag when starting to follow
+        SetDefaultRotation(); // Set default rotation when starting to follow
+        Logger.Information("Started following player {PlayerId}", playerId);
+    }
+
+    /// <summary>
+    /// Stops following the current player
+    /// </summary>
+    public void StopFollowingPlayer() {
+        IsFollowingPlayer = false;
+        FollowedPlayerId = null;
+        HasManuallyStopped = true; // Remember that user manually stopped
+        Logger.Information("Stopped following player");
+    }
+
+    /// <summary>
+    /// Unlocks camera follow when manual movement is detected
+    /// </summary>
+    public void UnlockCameraFollow() {
+        if (IsFollowingPlayer) {
+            IsFollowingPlayer = false;
+            Logger.Information("Camera follow unlocked - manual movement detected");
+        }
+    }
+
+    /// <summary>
+    /// Updates camera to follow a player at the specified position
+    /// </summary>
+    public void UpdatePlayerFollow(Vector3 playerPosition) {
+        if (!IsFollowingPlayer) return;
+
+        SetCameraTarget(playerPosition);
+        Camera.Transform.Position = playerPosition + CameraFollowOffset;
+        SetDefaultRotation(); // Always use default rotation when following player
+    }
+
+    /// <summary>
+    /// Sets the camera target and updates camera orientation
+    /// </summary>
+    public void SetCameraTarget(Vector3 target) {
+        CameraTarget = target;
+        Vector3 direction = Vector3.Normalize(target - Camera.Transform.Position);
+
+        // Calculate rotation to look at target
+        Vector3 up = Vector3.UnitZ; // Z is up in our coordinate system
+        Vector3 right = Vector3.Normalize(Vector3.Cross(direction, up));
+        up = Vector3.Cross(right, direction);
+
+        Matrix4x4 lookAtMatrix = new Matrix4x4(
+            right.X, up.X, -direction.X, 0,
+            right.Y, up.Y, -direction.Y, 0,
+            right.Z, up.Z, -direction.Z, 0,
+            0, 0, 0, 1
+        );
+
+        Camera.Transform.Transformation = lookAtMatrix;
+    }
+
+    /// <summary>
+    /// Sets the camera to the default rotation for MapleStory 2
+    /// </summary>
+    public void SetDefaultRotation() {
+        Camera.Transform.Quaternion = DefaultCameraRotation;
```

Comment:
Avoid touching quaternions when they're not needed. All of Maple2's stuff is already set up to use matrices so matrices should be used when possible (when doing anything that isn't skeletal animations or other sources of rotation interpolation).